### PR TITLE
Cleanup System.Threading.Tasks.Parallel.Tests, fix #790

### LINKED ISSUE #790: Refactor similar test into simpler helpers for ParallelFor tests
There's a lot of repetition in these tests that could easily be refactored into helpers.

see https://github.com/dotnet/corefx/blob/master/src/System.Threading.Tasks.Parallel/tests/ParallelFor.cs

### PR DESCRIPTION
**Pull request by hughbe**
Merge tests into single tests using [Theory] attributes
Fix minor formatting issues when I encountered them

Fixes #790

### CODE REVIEW

File: src/System.Threading.Tasks.Parallel/tests/ParallelFor.cs [hunk 1]
```diff
@@ -5,678 +5,71 @@
...
+        [InlineData(API.For64, StartIndexBase.Int32, 2, WithParallelOption.WithDOP, ActionWithState.Stop, ActionWithLocal.HasFinally)]
+        [InlineData(API.For64, StartIndexBase.Int32, 97, WithParallelOption.None, ActionWithState.None, ActionWithLocal.HasFinally)]
+        [InlineData(API.For64, StartIndexBase.Int32, 97, WithParallelOption.WithDOP, ActionWithState.Stop, ActionWithLocal.None)]
+
+        [InlineData(API.For, StartIndexBase.Zero, 0, WithParallelOption.None, ActionWithState.None, ActionWithLocal.None)]
+        [InlineData(API.For, StartIndexBase.Zero, 10, WithParallelOption.None, ActionWithState.None, ActionWithLocal.None)]
+        [InlineData(API.For, StartIndexBase.Zero, 10, WithParallelOption.None, ActionWithState.Stop, ActionWithLocal.HasFinally)]
+        [InlineData(API.For, StartIndexBase.Zero, 10, WithParallelOption.WithDOP, ActionWithState.None, ActionWithLocal.None)]
+        [InlineData(API.For, StartIndexBase.Zero, 1, WithParallelOption.None, ActionWithState.Stop, ActionWithLocal.None)]
+        [InlineData(API.For, StartIndexBase.Zero, 1, WithParallelOption.WithDOP, ActionWithState.Stop, ActionWithLocal.HasFinally)]
```
Comment (stephentoub) [inline src/System.Threading.Tasks.Parallel/tests/ParallelFor.cs]: It looks like this was ActionWithState.None in the original...?

File: src/System.Threading.Tasks.Parallel/tests/ParallelFor.cs [hunk 2]
```diff
@@ -5,678 +5,71 @@
...
+        [InlineData(API.ForeachOnArray, StartIndexBase.Zero, 1, WithParallelOption.WithDOP, ActionWithState.Stop, ActionWithLocal.None)]
+        [InlineData(API.ForeachOnArray, StartIndexBase.Zero, 2, WithParallelOption.None, ActionWithState.Stop, ActionWithLocal.None)]
+        [InlineData(API.ForeachOnArray, StartIndexBase.Zero, 2, WithParallelOption.WithDOP, ActionWithState.None, ActionWithLocal.HasFinally)]
+        [InlineData(API.ForeachOnArray, StartIndexBase.Zero, 97, WithParallelOption.None, ActionWithState.Stop, ActionWithLocal.HasFinally)]
+        [InlineData(API.ForeachOnArray, StartIndexBase.Zero, 97, WithParallelOption.WithDOP, ActionWithState.None, ActionWithLocal.None)]
+
+        [InlineData(API.ForeachOnList, StartIndexBase.Zero, 0, WithParallelOption.None, ActionWithState.None, ActionWithLocal.None)]
+        [InlineData(API.ForeachOnList, StartIndexBase.Zero, 10, WithParallelOption.None, ActionWithState.Stop, ActionWithLocal.HasFinally)]
+        [InlineData(API.ForeachOnList, StartIndexBase.Zero, 10, WithParallelOption.WithDOP, ActionWithState.Stop, ActionWithLocal.None)]
```
Comment (stephentoub) [inline src/System.Threading.Tasks.Parallel/tests/ParallelFor.cs]: In the original it looked like this with ActionWithState.None...?

File: src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs [hunk 3]
```diff
@@ -16,132 +16,93 @@ public static void RunParallelExceptionTests()
...
             // ParallelOptions tests
             ParallelOptions options = new ParallelOptions();
-            Assert.Throws<ArgumentOutOfRangeException>(
-               () => { options.MaxDegreeOfParallelism = 0; });
-            Assert.Throws<ArgumentOutOfRangeException>(
-               () => { options.MaxDegreeOfParallelism = -2; });
-
-            // 
+            Assert.Throws<ArgumentOutOfRangeException>(() => options.MaxDegreeOfParallelism = 0);
+            Assert.Throws<ArgumentOutOfRangeException>(() => options.MaxDegreeOfParallelism = -2);
```
Comment (stephentoub) [inline src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs]: As long as we're cleaning these up, can we validate the parameter name using the relevant Throws overload?

File: src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs [hunk 4]
```diff
@@ -349,132 +262,349 @@ public static void TestParallelForEachConversions()
...
         }
 
-        [Fact]
-        public static void RunSimpleParallelForeachAddTest_Enumerable()
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(1024)]
+        [InlineData(1024 * 1024)]
+        // This one just stopped working around 07/08/08 -- Can cause a horrible, indecipherable crash.
```
Comment (stephentoub) [inline src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs]: Let's get rid of this comment.  If there's actually a problem here, we'll find it and fix it.

File: src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs [hunk 5]
```diff
@@ -349,132 +262,349 @@ public static void TestParallelForEachConversions()
...
+         [InlineData(1024 * 1024 * 16)]
```
Comment (stephentoub) [inline src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs]: Nit: indentation

**Comment by stephentoub** [comment]
LGTM.  Thanks for cleaning these up, @hughbe.

**Comment by hughbe** [comment]
@stephentoub responded to your comments and rebased.

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**